### PR TITLE
feat/condo/infra-200/postgres-ha

### PR DESCRIPTION
**Pull request by Gladioluss**

### CODE REVIEW

**Comment by toplenboren** [comment]
Your branch looks a bit strange, I see some commits from @sitozzz , is this intentional?

**Comment by toplenboren** [comment]
Also, if you want review, the best practice is to leave a comment with description: what was done, why, and so on :-)

File: .github/workflows/_nodejs.condo.core.tests.yml [hunk 1]
```diff
@@ -24,31 +24,63 @@ jobs:
...
           POSTGRES_DB: main
 
-      - name: run redis for tests
+      - name: run redis
         run: |
           docker run -p="127.0.0.1:6379:6379" -d ${{ secrets.DOCKER_REGISTRY }}/doma/utils/redis:6.2
 
+      - name: run postgresql replica
+        run: |
+          docker run --name postgre-replica --user postgres --network=db_network -e PGUSER=$PGUSER -e PGPASSWORD=$PGPASSWORD -e POSTGRES_DB=$POSTGRES_DB -p="127.0.0.1:5433:5432" -d ${{ secrets.DOCKER_REGISTRY }}/doma/utils/postgres:13.2 bash -c "if [ ! -f /var/lib/postgresql/initialized ]; then pg_basebackup --pgdata=/var/lib/postgresql/data -R --slot=replication_slot --host=postgre-master --port=5432; chmod -R 0700 /var/lib/postgresql/data; touch /var/lib/postgresql/initialized; fi; postgres -c max_connections=2000"
```
Comment (pahaz) [inline .github/workflows/_nodejs.condo.core.tests.yml:64]: Why not just docket compose up -d ?

File: run_condo_domain_tests.sh [hunk 2]
```diff
@@ -51,7 +52,7 @@ node -e 'console.log(v8.getHeapStatistics().heap_size_limit/(1024*1024))'
 # NOTE(pahaz): Keystone not in dev mode trying to check dist/admin folder
 mkdir -p ./apps/condo/dist/admin
 
-[[ $DATABASE_URL == postgresql* ]] && yarn workspace @app/condo migrate
+[[ $DATABASE_URL == postgresql* || $DATABASE_URL == custom* ]] && yarn workspace @app/condo migrate
```
Comment (pahaz) [inline run_condo_domain_tests.sh:55]: Why not to run it always?

File: packages/keystone/databaseAdapters/ReplicaKnexAdapter.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+            let sql
+            let useMaster = true
+            try {
+                sql = builder.toSQL()
+                useMaster = Array.isArray(sql)
+                    ? checkUseMasterMultiple(sql)
+                    : checkUseMasterSingle(sql)
+            } catch (error) {
+                logger.error({
+                    err: error,
```
Comment (pahaz) [inline packages/keystone/databaseAdapters/ReplicaKnexAdapter.js:79]: May be we need to add sql to logging

File: .github/workflows/_nodejs.condo.core.tests.yml [hunk 4]
```diff
@@ -24,31 +24,63 @@ jobs:
           username: ${{ secrets.SBERCLOUD_CR_USERNAME }}
           password: ${{ secrets.SBERCLOUD_CR_PASSWORD }}
 
-      - name: run postgresql for tests
+      - name: run postgresql master
         run: |
-          docker run -e POSTGRES_USER=$POSTGRES_USER -e POSTGRES_PASSWORD=$POSTGRES_PASSWORD -e POSTGRES_DB=$POSTGRES_DB -p="127.0.0.1:5432:5432" -d ${{ secrets.DOCKER_REGISTRY }}/doma/utils/postgres:13.2
+          touch ./pg_hba.conf
```
[the remark this answers is not in the document]
Comment (sitozzz) [reply .github/workflows/_nodejs.condo.core.tests.yml:29]: That's not a main branch. Please add commits directly to the main task https://github.com/open-condo-software/condo/pull/4618

[the remark this answers is not in the document]
Comment (sitozzz) [reply .github/workflows/_nodejs.condo.core.tests.yml:29]: For now self-hosted runner is unable to run compose services. In master branch we already using same service deployment strategy, so your comment is unrelated to this PR. Runner will be updated and after that I'll fix that

**Comment by sitozzz** [comment]
Close this PR due to duplication of the original one https://github.com/open-condo-software/condo/pull/4618

**Pull request closed without merging**